--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -40,11 +40,14 @@ SET(BUNDLER_SOURCES BaseApp.cpp BundlerA
 	BoundingBox.cpp BundleAdd.cpp ComputeTracks.cpp BruteForceSearch.cpp
 	BundleIO.cpp ProcessBundle.cpp BundleTwo.cpp Decompose.cpp
 	RelativePose.cpp Distortion.cpp TwoFrameModel.cpp LoadJPEG.cpp)
+
 SET_SOURCE_FILES_PROPERTIES(${BUNDLER_SOURCES}
   PROPERTIES
   COMPILE_FLAGS "-D__NO_UI__ -D__BUNDLER__ -D__BUNDLER_DISTR__ -D_CRT_SECURE_NO_WARNINGS")
+
 ADD_EXECUTABLE(Bundler ${BUNDLER_SOURCES})
 TARGET_LINK_LIBRARIES(Bundler imagelib sfm-driver sba-1.5 matrix -lz
  5point ${JPEG_LIBRARY} ann_1.1_char getopt ${MATH_LIBS})
 
+INSTALL(TARGETS Bundler Bundle2PMVS Bundle2Vis RadialUndistort KeyMatchFull RUNTIME DESTINATION bin)
 set(CMAKE_VERBOSE_MAKEFILE ON)
